go/vt/discovery/tablet_picker_test.go

```diff
@@ -364,6 +364,108 @@ func TestPickUsingCellAlias(t *testing.T) {
 	assert.True(t, picked2)
 }
 
+func TestPickLocalPreference(t *testing.T) {
```

Comment:
Since the tests are very similar, you can combine all the three tests into a single table-based one. For example, see `TestVStreamSkew` or `TestGTIDCompress` . Essentially you can define a struct type with test options, inputs and expected outputs and iterate over it with conditional logic.

Reply:
I will do this for the tests introduced in `vstream_manager_test.go`. The ones here are a bit more complex since they have different number of tablet, types, cells in each. It's becoming more complex to merge these.